### PR TITLE
Getting the notify api key from secrets

### PR DESCRIPTION
**Pull request by szabikr**

### CODE REVIEW

File: src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts [hunk 1]
```diff
@@ -42,6 +43,26 @@ export class ConfigAdapter implements IConfigAdapter {
...
+    if (this.isLocal) {
+      this.apiKey = this.getFromEnvThrowIfNotPresent('NOTIFY_API_KEY');
+      return;
+    }
+
+    const secretsmanager = new awsSdk.SecretsManager();
+    const params: awsSdk.SecretsManager.GetSecretValueRequest = {
+      SecretId: 'mes-nonprod-api-documents-service', // TODO: Confirm that this is really the name of the secret
+    };
+    secretsmanager.getSecretValue(params, (err, data) => {
```
Comment (ammarhaiderbjss) [inline src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts]: no `.promise()` available on `.getSecretValue()`?

Comment (szabikr) [reply src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts]: yes, there is

Comment (ammarhaiderbjss) [reply src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts]: be good to use `await` then to stay consistent

File: src/functions/sendCandidateResults/framework/di/inversify.config.ts [hunk 2]
```diff
@@ -24,7 +24,9 @@ container.bind<IConfigAdapter>(TYPES.IConfigAdapter).to(ConfigAdapter);
 
 const configAdapter: IConfigAdapter = container.get<IConfigAdapter>(TYPES.IConfigAdapter);
 
-container.bind<string>(TYPES.apiKey).toConstantValue(configAdapter.apiKey);
+configAdapter.getApiKey()
+  .then(apiKey => container.bind<string>(TYPES.apiKey).toConstantValue(apiKey))
+  .catch(err => container.bind<string>(TYPES.apiKey).toConstantValue(''));
```
Comment (ammarhaiderbjss) [inline src/functions/sendCandidateResults/framework/di/inversify.config.ts]: Should we not throw an error when this happens?

Comment (szabikr) [reply src/functions/sendCandidateResults/framework/di/inversify.config.ts]: Yes you are right. Shouldn't swallow the error

File: src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts (hunk 1 above)
Comment (szabikr) [reply src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts]: changed it

File: src/functions/sendCandidateResults/framework/di/inversify.config.ts (hunk 2 above)
Comment (szabikr) [reply src/functions/sendCandidateResults/framework/di/inversify.config.ts]: logging out error if there is no api key

File: src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts [hunk 3]
```diff
@@ -42,6 +45,31 @@ export class ConfigAdapter implements IConfigAdapter {
...
+    const documentServiceSecretName = this.getFromEnvThrowIfNotPresent('DOCUMENTS_SERVICE_SECRET_NAME');
+
+    const secretsmanager = new awsSdk.SecretsManager();
+    const params: awsSdk.SecretsManager.GetSecretValueRequest = {
+      SecretId: documentServiceSecretName,
+    };
+
+    try {
+      const secretValue = await secretsmanager.getSecretValue(params).promise();
+      const secrets = JSON.parse(secretValue.SecretString || '');
```
Comment (ghost) [inline src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts:66]: If the default empty string is used here, you will immediately get a NPE on the next line.

**Review by rossfauldsbjss — APPROVED** [review]
looks good.

**Review by ghost — APPROVED** [review]
I like how you're grabbing the API key and binding to a constant value in Inversify. One minor thing to handle errors a bit more gracefully, but I think it's already covered by your existing catches.

**Review by ammarhaiderbjss — APPROVED** [review]

File: src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts (hunk 3 above)
Comment (szabikr) [reply src/functions/sendCandidateResults/framework/adapter/config/config-adapter.ts:66]: And the next line after will catch it and reject the procedure. Wouldn't that be a desired outcome?

**Pull request merged**